src/composables/accountSelector.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  computed,
+  ref,
+  watch,
+} from 'vue';
+
+import type {
+  IAccountSelectorEntry,
+  ITransaction,
+} from '@/types';
+import {
+  ACCOUNT_SELECT_TYPE_FILTER,
+  AccountSelectTypeFilter,
+  PROTOCOLS,
+  TX_DIRECTION,
+} from '@/constants';
+import {
+  createCustomScopedComposable,
+  excludeFalsy,
+  getDefaultAccountLabel,
+  getDirection,
+} from '@/utils';
+import {
+  useAccounts,
+  useAddressBook,
+  useLatestTransactionList,
+} from '@/composables';
+
+import { useAeNames } from '@/protocols/aeternity/composables/aeNames';
+import { getInnerTransaction, getOwnershipStatus, getTxOwnerAddress } from '@/protocols/aeternity/helpers';
+
+import { tg } from '@/popup/plugins/i18n';
+
+export const useAccountSelector = createCustomScopedComposable(() => {
+  const { getName } = useAeNames();
+
+  const searchQuery = ref('');
+  const {
+    addressBookFiltered: rawAddressBookFiltered,
+    addressBookFilteredByProtocol,
+    protocolFilter,
+    showBookmarked,
+    getAddressBookEntryByAddress,
+    setProtocolFilter,
+    setShowBookmarked,
+    clearFilters: addressBookClearFilters,
+  } = useAddressBook();
+  const {
+    activeAccount,
+    accounts,
+    accountsGroupedByProtocol,
+    getAccountByAddress,
+  } = useAccounts();
+  const { accountsTransactionsLatest } = useLatestTransactionList();
+  const accountSelectType = ref<AccountSelectTypeFilter>(ACCOUNT_SELECT_TYPE_FILTER.addressBook);
+  const addressBookFiltered = computed(() => rawAddressBookFiltered.value.map((account) => ({
+    ...account,
+    isAddressBookEntry: true,
+  })));
+  const latestTransactions = computed(
+    () => (accountsTransactionsLatest.value[activeAccount.value.address] || [])
+      .map((transaction: ITransaction): IAccountSelectorEntry | null => {
+        const outerTx = transaction.tx!;
+        const innerTx = transaction.tx ? getInnerTransaction(transaction.tx) : null;
+        const { recipientId } = outerTx?.payerId ? outerTx : innerTx;
+        const direction = getDirection({
+          innerTx,
+          outerTx,
+          ownershipStatus: getOwnershipStatus(activeAccount.value, accounts.value, innerTx),
+          txOwnerAddress: getTxOwnerAddress(innerTx),
+          activeAccountAddress: activeAccount.value.address,
+        });
+        if (direction !== TX_DIRECTION.sent || !recipientId) {
+          return null;
+        }
+
+        let name = tg('modals.send.recipientLabel');
+        const isChainName = recipientId?.startsWith('nm_');
```

Comment:
I suggest to rename this to something like `isSentToChainName` because `isChainName` doesn't fully reflect the reason for the logic below. What do you think?